docs/database-classes.md

```diff
@@ -1,348 +1,283 @@
-
-## Limitations
-
-### Property limit
-
-Database classes can have a maximum of 112 properties for performance reasons. The limit applies to the total number of persistent properties \(including all inherited\) per class.
-
-Thus, this is not allowed:
-
-```csharp
-[Database]
-public class LargeClass
-{
-    public virtual string Property1 { get; set; }
-    public virtual string Property2 { get; set; }
-
-    // ...
-
-    public virtual string Property113 { get; set; }
+    public abstract string Name { get; set; }
 }
 ```
 
-If a database class has more than 113 properties, Starcounter throws `ScErrToManyAttributes (SCERR4013)`.
-
-### Nested classes
-
-Nested database classes are not supported. The limitation is that inner database classes cannot be queried with SQL.
+**Note**: DDL statements can only be performed from `IDdlExecutor`.
 
 ## Relations
 
 ### One-to-many relations
 
-We recommend modeling one-to-many relationships by having references both ways - the child has a reference to the parent and the parent has a method to select all the children.
-
-In this example there is a one-to-many relationship between `Department` and `Employee`:
-
-```csharp
-using Starcounter.Database;
-
-[Database]
-public class Department
-{
-    public IEnumerable<Employee> SelectEmployees(IDatabaseContext db)
-    {
-        return db.Sql<Employee>
-        (
-            "SELECT e FROM Employee e WHERE e.Department = ?",
-            this
-        );
-    }
-}
-
-[Database]
-public class Employee
-{
-    public abstract Department Department { get; set; }
-}
-```
+It's recommended to model one-to-many relations using references both ways, with the child entity having a reference to the parent and the parent having an instance method or computed property that selects all the children (like in the example with the `Mother`/`Children` relation above).
 
 ### Many-to-many relations
 
-We recommend modeling many-to-many relationships with an associative class.
+Many-to-many relations are best modelled using an **association class**.
 
-In this example there is a many-to-many relation between `Person` and `Company` - to represent this many-to-many relationship we use the associative class `Shares`:
+Let's say we want to model many-to-many relation of **share ownership**, as understood as a relation between `Person` and `Company` entities such that a single person can hold multiple shares in multiple companies, and a single company can have multiple shareholders. For this, let's use the association class `HoldsShares`. We can then make queries to this database class to calculate useful information in `Person` and `Company` that is exposed using computed properties. Lastly, we make sure that indexes are registered for the `Owner` and `Equity` properties of `HoldsShares` since we will be quering them a lot.
```

Comment:
`Shareholder` should be a better name for this table, instead of `HoldsShares`.

Reply:
A shareholder is a relation between a person and a company, with quantity?

Reply:
I would still call it just a `Shareholder` with quantity :), but let's keep it as is.

Reply:
But it's not a shareholder. It's the relation between a shareholder and the shares he or she holds. You are not a Starcounter employment, you're an employee :)

Reply:
But MongoDB is web scaled ©.

>It's the relation between a shareholder and the shares he or she holds.

It's a relation between holder & shares. Having such a relation with quantity > 0 makes a holder to be a shareholder :).

Whatever, let's keep it your way, I am fine with it, just thought `Shareholder` is a neat word.